### PR TITLE
Vagrant: Update to box version 52 and enable core dumps

### PR DESCRIPTION
**Pull request by jrajahalme**
Enable core dumps both in dev VM and CI VMs, and update `cilium.service` to not limit cores.

This does not change CI test behavior, but enables collecting cores manually from local CI runs.

Signed-off-by: Jarno Rajahalme <jarno@covalent.io>

### CODE REVIEW

**Comment by jrajahalme** [comment]
This overlaps with #3355, but was still very useful for local debugging without changing CI behavior.

File: test/provision/runtime_install.sh [hunk 1]
```diff
@@ -17,5 +17,9 @@ cd ${GOPATH}/src/github.com/lyft/protoc-gen-validate
 sudo git checkout 930a67cf7ba41b9d9436ad7a1be70a5d5ff6e1fc
 make build
 
+sudo systemctl disable apport.service
```
Comment (eloycoto) [inline test/provision/runtime_install.sh]: Could you update the VM to version 52 and you don't need to add these ones. 

https://app.vagrantup.com/cilium/boxes/ubuntu/versions/52

**Review by eloycoto — CHANGES_REQUESTED** [review]

**Comment by jrajahalme** [comment]
test-me-please

**Comment by jrajahalme** [comment]
Updated to use the newer box (52) as requested.

**Comment by eloycoto** [comment]
LGTM

**Comment by jrajahalme** [comment]
test-me-please

**Comment by ianvernon** [comment]
While approval was not given, merging since @eloycoto said LGTM, which equates to an approval.

**Pull request merged**